payment_wcpay/controllers/main.py

```diff
@@ -0,0 +1,425 @@
+
+        wc_return = Wxpay_server_pub()
+        wc_return.setReturnParameter('return_code', return_code)
+        wc_return.setReturnParameter('return_msg', return_msg)
+        return_xml = wc_return.returnXml()
+        return return_xml
+
+    @http.route(['/shop/confirmation',
+                 '/shop/confirmation/so/<int:sale_order_id>',
+                 '/shop/confirmation/ai/<int:account_invoice_id>'],
+                type='http', auth="public", website=True)
+    def payment_confirmation(self, sale_order_id=None, account_invoice_id=None, **post):
+        cr, uid, context = request.cr, request.uid, request.context
+
+        if post.get('subject'):
+            subject = post.get('subject')
+            sale_order_ids = request.registry['sale.order'].search(
+                cr, uid, [('name', '=', subject)], context=context
+            )
+            if sale_order_ids:
+                sale_order_id = sale_order_ids[0]
+
+            if sale_order_id is None:
+                account_invoice_ids = request.registry['account.invoice'].search(
+                    cr, uid, [('number', '=', subject)], context=context
+                )
+                if account_invoice_ids:
+                    account_invoice_id = account_invoice_ids[0]
+
+        if sale_order_id is None and \
+                        account_invoice_id is None and \
+                request.session.uid and \
+                request.session.get('sale_last_order_id'):
+            sale_order_id = request.session.get('sale_last_order_id')
+
+        acquirer = request.registry['payment.acquirer'].search(
+            cr, uid, [('name', '=', 'Wechat Pay')])
+
+        if sale_order_id:
+            order = request.registry['sale.order'].browse(
+                cr, SUPERUSER_ID, sale_order_id, context=context)
+            website_sale.payment_transaction_for_so(website_sale(), acquirer[0], order.id)
+            return request.website.render("website_sale.confirmation", {'order': order})
+        elif account_invoice_id:
+            invoice = request.registry['account.invoice'].browse(
+                cr, SUPERUSER_ID, account_invoice_id, context=context)
+            website_sale.payment_transaction_for_ai(website_sale(), acquirer[0], invoice.id)
+            return request.website.render("payment_wcpay.wcpay_confirmation", {'invoice': invoice})
+        else:
+            return request.redirect('/shop')
+
+    @http.route('/shop/payment/get_status_so/<int:sale_order_id>', type='json', auth="public", website=True)
+    def payment_get_status_so(self, sale_order_id, **post):
+        cr, uid, context = request.cr, request.uid, request.context
+        order = request.registry['sale.order'].browse(cr, SUPERUSER_ID, sale_order_id, context=context)
+        assert order.id == sale_order_id
+
+        if not order:
+            return {
+                'state': 'error',
+                'message': '<p>%s</p>' % _('There seems to be an error with your request.'),
```

Comment:
Can we improve the Error type?
With current message we have no way to know what went wrong.